sh/drive-utils.sh

```diff
@@ -0,0 +1,535 @@
+                _file_id_clone_file="$(printf "%s\n" "${existing_file_check_json_clone_file}" | _json_value id 1 1)"
+                post_data_clone_file="$(_drive_info "${_file_id_clone_file}" "parents,writersCanShare" "${token_clone_file}")"
+                if [ "${_file_id_clone_file}" != "${file_id_clone_file}" ]; then
+                    curl --compressed -s \
+                        -X DELETE \
+                        -H "Authorization: Bearer ${token_clone_file}" \
+                        "${API_URL}/drive/${API_VERSION}/files/${_file_id_clone_file}?supportsAllDrives=true" 2> /dev/null 1>&2 || :
+                    STRING="Updated"
+                else
+                    _collect_file_info "${existing_file_check_json_clone_file}" "${name_clone_file}" || return 1
+                fi
+            fi
+        else
+            _print_center "justify" "Cloning file.." "-"
+        fi
+    else
+        _print_center "justify" "Cloning file.." "-"
+    fi
+
+    # shellcheck disable=SC2086
+    response_clone_file="$(curl --compressed ${CURL_PROGRESS} \
+        -X POST \
+        -H "Authorization: Bearer ${token_clone_file}" \
+        -H "Content-Type: application/json; charset=UTF-8" \
+        -d "${post_data_clone_file}" \
+        "${API_URL}/drive/${API_VERSION}/files/${file_id_clone_file}/copy?supportsAllDrives=true" || :)"
+    for _ in 1 2 3; do _clear_line 1; done
+    if [ -n "${response_clone_file}" ]; then
+        _collect_file_info "${response_clone_file}" "${name_clone_file}" || return 1
+        "${QUIET:-_print_center}" "justify" "${name_clone_file} " "| ${readable_size_clone_file} | ${STRING}" "="
+    else
+        _error_logging "${name_clone_file}" && return 1
+    fi
+    return 0
+}
+
+###################################################
+# Share a gdrive file/folder
+# Globals: 2 variables, 4 functions
+#   Variables - API_URL and API_VERSION
+#   Functions - _url_encode, _json_value, _print_center, _clear_line
+# Arguments: 3
+#   ${1} = gdrive ID of folder/file
+#   ${2} = Access Token
+#   ${3} = Email to which file will be shared ( optional )
+# Result: read description
+# Reference:
+#   https://developers.google.com/drive/api/v3/manage-sharing
+###################################################
+_share_id() {
+    [ $# -lt 2 ] && printf "Missing arguments\n" && return 1
+    id_share_id="${1}" token_share_id="${2}" share_email_share_id="${3}" role_share_id="reader" type_share_id="anyone"
+    unset post_data_share_id response_share_id
+
+    "${EXTRA_LOG:-:}" "justify" "Sharing.." "-" 1>&2
+
+    if [ -n "${share_email_share_id}" ]; then
+        type_share_id="user"
+        post_data_share_id="{\"role\":\"${role_share_id}\",\"type\":\"${type_share_id}\",\"emailAddress\":\"${share_email_share_id}\"}"
+    else
+        post_data_share_id="{\"role\":\"${role_share_id}\",\"type\":\"${type_share_id}\"}"
+    fi
+
+    response_share_id="$(curl --compressed ${CURL_PROGRESS_EXTRA} \
```

Comment:
Codacy found an issue: [Double quote to prevent globbing and word splitting.](https://app.codacy.com/manual/labbots/google-drive-upload/pullRequest?prid=5865418)